auth_session_timeout/models/ir_config_parameter.py

```diff
@@ -14,23 +14,34 @@ class IrConfigParameter(models.Model):
     _inherit = 'ir.config_parameter'
 
     @tools.ormcache(skiparg=0)
-    def get_session_parameters(self, db):
+    def _auth_timeout_get_parameter_delay(self, db):
         param_model = self.pool['ir.config_parameter']
         cr = self.pool.cursor()
```

Comment:
I wasn't sure why this was done originally, so followed the pattern in case it had good reason.

This is all complete conjecture, but:

Seems to me that the `ResUser.check` method could quite likely be called on requests that are at a stage that have not had their proper environment/context set up yet as part of request buildup/teardown.

I.e. Odoo should check a users validation credentials before opening them an active db cursor, etc.

Since this module hangs off `ResUser.check`, we get caught in that scenario.

The only scenario we are actually interested in though is (re)checks - we don't need to check for session expiry on the request that is logging a user into the system (but we do)

It would not surprise me if the manual cursor control was added as a workaround by the original developer(s) as they quickly realized that without them users could no longer log in!